tests/Integration/Behaviour/Features/Scenario/Cart/Calculation/CartRule/free_shipping.feature

```diff
@@ -0,0 +1,28 @@
+@reset-database-before-feature
+Feature: Cart rule (amount) calculation with one cart rule offering free shipping
+  As a customer
+  I must be able to have correct cart total when adding cart rules
+
+  Scenario: One product in cart, one cartRule offering only free shipping
+    Given I have an empty default cart
+    Given shop configuration for "PS_CART_RULE_FEATURE_ACTIVE" is set to 1
+    Given there is a product in the catalog named "product1" with a price of 19.812 and 1000 items in stock
+    Given there is a cart rule named "cartrule4" that applies no discount with priority 4, quantity of 1000 and quantity per user 1000
+    Given cart rule "cartrule4" offers free shipping
+    Given cart rule "cartrule4" has a discount code "foo4"
+    When I add 1 items of product "product1" in my cart
+    When I use the discount "cartrule4"
```

Comment:
Shouldn't it be `I use the discount code "foo4"`

Reply:
it could be both, here the code doesn't matter, the only thing that matter is that the voucher `has` a code (not applied anytime)